### PR TITLE
Hosting Command Palette: Add more site management commands

### PR DESCRIPTION
**Pull request by danielbachhuber**
See https://github.com/Automattic/dotcom-forge/issues/4705
Related https://github.com/Automattic/jetpack-commands-raycast/pull/29

## Proposed Changes

Adds another set of site management commands:
* Add Jetpack to a self-hosted site
* Import site to WordPress.com
* Install theme
* Install plugin
* Manage general settings

Also ensures the other settings commands appropriately respect 'wp-admin' site default.

CleanShot 2023-12-03 at 08 50 54@2x

## Testing Instructions

Test the commands and verify they work as expected.

### CODE REVIEW

File: client/sites-dashboard/components/wpcom-smp-commands.tsx [hunk 1]
```diff
@@ -569,6 +600,24 @@ export const useCommandsArrayWpcom = ( {
 			},
 			icon: brushIcon,
 		},
+		{
+			name: 'installTheme',
+			label: __( 'Install theme' ),
```
Comment (wojtekn) [inline client/sites-dashboard/components/wpcom-smp-commands.tsx:784]: What is the benefit of having this one if we have "Manage themes"?

File: client/sites-dashboard/components/wpcom-smp-commands.tsx [hunk 2]
```diff
@@ -587,6 +636,24 @@ export const useCommandsArrayWpcom = ( {
 			},
 			icon: pluginsIcon,
 		},
+		{
+			name: 'installPlugin',
+			label: __( 'Install plugin' ),
```
Comment (wojtekn) [inline client/sites-dashboard/components/wpcom-smp-commands.tsx:833]: The question about the "Install theme" applies here too.

In my scenario, I test both the "Install plugin" and "Manage plugins" actions and they point to the same page.

File: client/sites-dashboard/components/wpcom-smp-commands.tsx [hunk 3]
```diff
@@ -675,6 +742,27 @@ export const useCommandsArrayWpcom = ( {
...
+			callback: setStateCallback(
+				'manageSettingsGeneral',
+				__( 'Select site to manage general settings' )
+			),
+			siteFunctions: {
+				onClick: ( { site, close }: { site: SiteExcerptData; close: () => void } ) => {
+					close();
+					const link =
+						( site.jetpack && ! site.is_wpcom_atomic ) ||
+						'wp-admin' === site.options?.wpcom_admin_interface
```
Comment (wojtekn) [inline client/sites-dashboard/components/wpcom-smp-commands.tsx:975]: Would it make sense to add an issue to abstract those?

File: client/sites-dashboard/components/wpcom-smp-commands.tsx (hunk 1 above)
Comment (danielbachhuber) [reply client/sites-dashboard/components/wpcom-smp-commands.tsx:784]: "Manage themes" and "Install themes" go to two different places in "wp-admin".

File: client/sites-dashboard/components/wpcom-smp-commands.tsx (hunk 3 above)
Comment (danielbachhuber) [reply client/sites-dashboard/components/wpcom-smp-commands.tsx:975]: Maybe, but not in this PR

File: client/sites-dashboard/components/wpcom-smp-commands.tsx (hunk 1 above)
Comment (wojtekn) [reply client/sites-dashboard/components/wpcom-smp-commands.tsx:784]: Yes, but they go to the same place when Calypso is used.

File: client/sites-dashboard/components/wpcom-smp-commands.tsx (hunk 2 above)
Comment (wojtekn) [reply client/sites-dashboard/components/wpcom-smp-commands.tsx:833]: If we are going to keep both, should we add some keywords, e.g., "add a plugin" and maybe "plugins"?

Comment (danielbachhuber) [reply client/sites-dashboard/components/wpcom-smp-commands.tsx:833]: Sure, added with a717178c865

File: client/sites-dashboard/components/wpcom-smp-commands.tsx (hunk 3 above)
Comment (sejas) [reply client/sites-dashboard/components/wpcom-smp-commands.tsx:975]: - Created https://github.com/Automattic/dotcom-forge/issues/4764

**Review by sejas — APPROVED** [review]
I tested the commands on atomic and Jetpack sites. They worked as expected. My only concern is, what @wojtekn  already [said]( https://github.com/Automattic/wp-calypso/pull/84791/files#r1413685837) that manage plugins/themes and install plugins/themes will conflict. And both are almost identical.

https://github.com/Automattic/wp-calypso/assets/779993/581cc9c9-70e8-46ab-8431-3617b5d7a5ab

**Comment by sejas** [comment]
@javierarce , if you have any  feedback, like icons. We can open a follow-up PR.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/9878248

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @danielbachhuber for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/9878248).

**Pull request merged**